superset-frontend/src/dashboard/components/nativeFilters/FilterBar/FilterControls/FilterDivider.tsx

```diff
@@ -0,0 +1,191 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { css, SupersetTheme } from '@superset-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+import Icons from 'src/components/Icons';
+import { Tooltip } from 'src/components/Tooltip';
+
+export interface FilterDividerProps {
+  title: string;
+  description: string;
+  horizontal?: boolean;
+  horizontalOverflow?: boolean;
+}
+
+const useIsTruncated = <T extends HTMLElement>(
+  text: string,
+): [React.RefObject<T>, boolean] => {
+  const ref = useRef<T>(null);
+  const [isTruncated, setIsTruncated] = useState(true);
+  useEffect(() => {
+    if (ref.current) {
+      setIsTruncated(ref.current.offsetWidth < ref.current.scrollWidth);
+    }
+  }, [text]);
+
+  return [ref, isTruncated];
+};
+
+const VerticalDivider = ({ title, description }: FilterDividerProps) => (
+  <div>
+    <h3>{title}</h3>
+    {description ? <p data-test="divider-description">{description}</p> : null}
+  </div>
+);
+
+const HorizontalDivider = ({ title, description }: FilterDividerProps) => {
+  const [titleRef, titleIsTruncated] =
+    useIsTruncated<HTMLHeadingElement>(title);
+
+  const tooltipOverlay = (
+    <>
+      {titleIsTruncated ? (
+        <div>
+          <strong>{title}</strong>
+        </div>
+      ) : null}
+      {description ? <div>{description}</div> : null}
+    </>
+  );
+
+  return (
+    <div
+      css={(theme: SupersetTheme) => css`
+        display: flex;
+        align-items: center;
+        height: ${8 * theme.gridUnit}px;
+        border-left: 1px solid ${theme.colors.grayscale.light2};
+        padding-left: ${4 * theme.gridUnit}px;
+        margin-right: ${4 * theme.gridUnit}px;
+      `}
+    >
+      <h3
+        ref={titleRef}
+        css={(theme: SupersetTheme) => css`
+          max-width: 130px;
+          white-space: nowrap;
+          overflow: hidden;
+          text-overflow: ellipsis;
+          font-size: 14px;
+          font-weight: normal;
+          margin: 0;
+          color: ${theme.colors.grayscale.dark1};
+        `}
+      >
+        {title}
+      </h3>
+      {titleIsTruncated || description ? (
+        <Tooltip overlay={tooltipOverlay}>
+          <Icons.BookOutlined
+            data-test="divider-description-icon"
+            css={(theme: SupersetTheme) => css`
+              color: ${theme.colors.grayscale.base};
+              font-size: 16px;
```

Comment:
Icons have a property `iconSize` with predefined sizes. We should stick to those whenever possible.

Reply:
Done!